nomos-services/data-availability/src/backend/mod.rs

```diff
@@ -0,0 +1,14 @@
+#[derive(Debug)]
+pub enum DaError {}
+
+pub trait DaBackend {
+    type Settings: Clone;
+
+    type Blob;
+
+    fn new(settings: Self::Settings) -> Self;
+
+    fn add_blob(&mut self, blob: Self::Blob) -> Result<(), DaError>;
+
+    fn pending_blobs(&self) -> Box<dyn Iterator<Item = Self::Blob> + Send>;
```

Comment:
the expectation is that pending blobs are removed from the list once returned? or once polled from the iterator? or is that signaled externally?

Reply:
The idea is to remove the blobs in the same fashion we do in the mempool actually. We can use some evicting policy for really old ones as well.

Reply:
I don't see any method to change the blobs status though here

Reply:
Yes. Ill add it in the incoming backend PR.